adventurous-anteaters/adv_ant/retroNet/views.py

```diff
@@ -0,0 +1,106 @@
+from django.contrib.auth import login
+from django.http import HttpResponse
+from django.shortcuts import render, redirect
+from django.contrib.auth.forms import UserCreationForm
+from .models import Tweet, UpdateProfile
+from .forms import TweetForm, Updateprofile
+from django.contrib.auth.decorators import login_required
+from django.contrib.auth import logout
+
+
+# index view when user logs in
+
+@login_required(login_url='/login/')
+def createpost(request):
+    return render(request, 'create_tweet.html')
+
+
+@login_required(login_url='/login/')
+def index(request):
+    form = TweetForm(request.POST or None)
+    if form.is_valid():
+        form.save()
+        queryset = Tweet.objects.all()
+        return render(request, 'index.html', {"data": queryset})
+    else:
+        queryset = Tweet.objects.all()
+        return render(request, 'index.html', {"data": queryset})
+
+
+# for registration of a new user
+def register(request):
+    context = {}
+    form = UserCreationForm(request.POST or None)
+    if request.method == "POST":
+        if form.is_valid():
+            user = form.save()
+            login(request, user)
+            return render(request, 'tweet.html')
+    context['form'] = form
+    return render(request, 'register.html', context)
+
+
+@login_required(login_url='/login/')
+def update_profile(request):
+    return render(request, 'temp_test.html')
+
+
+@login_required(login_url='/login/')
+def update_profile_display(request):
+    form = Updateprofile(request.POST or None)
+    if form.is_valid():
+        form.save()
+        querysets = UpdateProfile.objects.all()
+        # TODO add proper templates in both return
```

Comment:
don't leave TODO's in your code - TODO comments is where ideas go to die. Make TODOs for yourself, and then address them before you open your pull request to `master`.